openfl/experimental/runtime/local_runtime.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (C) 2020-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+""" openfl.experimental.runtime package LocalRuntime class."""
+
+from copy import deepcopy
+import ray
+import gc
+from openfl.experimental.runtime import Runtime
+from openfl.experimental.placement import RayExecutor
+from openfl.experimental.utilities import (
+    aggregator_to_collaborator,
+    generate_artifacts,
+    filter_attributes,
+    checkpoint,
+)
+
+
+class LocalRuntime(Runtime):
+    def __init__(
+        self,
+        aggregator=None,
+        collaborators=None,
+        backend="single_process",
+        **kwargs,
+    ):
+        """Use Local infrastructure to run the flow"""
+        super().__init__()
+        if backend not in ["ray", "single_process"]:
```

Comment:
In my understanding, `ray` and `single_process` backends may be classified as separate runtimes. Does it make sense to have separate  `RayLocalRuntime` and `SPLocalRuntime` classes both implementing the `Runtime` interface?

Reply:
@igor-davidyuk I'm glad you raised this. My intention was to create a hierarchy and logically divide `Runtime` to define where the workflow would run, and the `Backend` would specify how it would be executed. That way in the future there it will be clear if the user selects `LocalRuntime` the workflow will be executed locally, and `FederatedRuntime` runs the workflow across multiple nodes, and the user doesn't need to think about the backend unless they have good reason to do so. @mansishr, @aleksandr-mokrov, @itrushkin what are your thoughts about this?

Reply:
Suppose we want to differentiate between runtimes and backends. In that case, we need a `backend` interface class to encapsulate the backend methods. 
`if else` approach is not sustainable if we would add more backends.

Reply:
I like the idea of dividing `Runtime` to define where the flow would run, making it clear for the user. About `Backend` being an interface class, for now we only support `Ray` so using `if else` looks okay to me.

Reply:
@igor-davidyuk and I discussed this offline. I agree a formal interface for the backend is more sustainable. I've added an issue for this to be addressed in the next release: https://github.com/intel/openfl/issues/676